starters/graphql-serverless-contentful/src/schema/post/post.resolver.test.ts

```diff
@@ -0,0 +1,204 @@
+import gql from 'graphql-tag';
+import { apolloServer } from '../../handlers/graphql';
+import { GraphQLResponse } from '@apollo/server/src/externalTypes';
+import assert from 'assert';
+
+const MOCK_CONTENT = [
+	'This is the test post content',
+	'This is the updated test post content',
+];
+
+jest.mock('../../utils/contentful', () => {
+	const MOCK_CONTENT = [
+		'This is the test post content',
+		'This is the updated test post content',
+	];
+	const MOCK_ID = 'xxx';
+	const MOCK_CONTENTFUL_ENTRIES = {
+		items: MOCK_CONTENT.map((content) => ({
+			sys: {
+				id: MOCK_ID,
+			},
+			fields: {
+				content: {
+					'en-US': content,
+				},
+			},
+		})),
+	};
+
+	const MOCK_CONTENTFUL_ENVIRONMENT = {
+		getEntry: jest.fn().mockReturnValue({
+			...MOCK_CONTENTFUL_ENTRIES.items[0],
+			publish: jest.fn().mockReturnValue({}),
+			update: jest.fn().mockReturnValue({}),
+			updatePost: {
+				id: MOCK_ID,
+				content: MOCK_CONTENT[1],
+			},
+		}),
+		createEntry: jest.fn().mockReturnValue({
+			...MOCK_CONTENTFUL_ENTRIES.items[0],
+			publish: jest.fn().mockReturnValue({
+				createPost: {
+					id: MOCK_ID,
+					content: MOCK_CONTENT[0],
+				},
+			}),
+		}),
+		getEntries: jest.fn().mockReturnValue(MOCK_CONTENTFUL_ENTRIES),
+	};
+	const MOCK_CONTENTFUL_SPACE = {
+		getEnvironment: jest.fn().mockReturnValue(MOCK_CONTENTFUL_ENVIRONMENT),
+	};
+	const MOCK_CONTENTFUL_CLIENT = {
+		client: {
+			getSpace: jest.fn().mockReturnValue(MOCK_CONTENTFUL_SPACE),
+		},
+	};
+	return MOCK_CONTENTFUL_CLIENT;
+});
+
+describe('post queries and mutations', () => {
+	let post: any;
+	let content: string;
+
+	beforeAll(async () => {
+		content = 'This is the test post content';
+		const MUTATION = gql`
+			mutation CreatePostMutation($content: String!) {
+				createPost(content: $content) {
+					id
+					content
+				}
+			}
+		`;
+
+		const res: GraphQLResponse = await apolloServer.executeOperation({
+			query: MUTATION,
+			variables: {
+				content,
+			},
+		});
+
+		assert(res.body.kind === 'single');
+		expect(res.body.singleResult.errors).toBeUndefined();
+		assert(res.body.singleResult.data);
+
+		post = res.body.singleResult.data.createPost;
+	});
```

Comment:
I might have only discussed this with @iansamz before you joined the call. The mutation is not required for the other tests to succeed. Seeing we have the persistence of data completely mocked, it wouldn't store the result of this mutation anyway. We can safely wrap it in its own test without it having to be `beforeAll`.

All other tests would continue to work, as the mock contentful will return the default posts and comment. We just need to refer to the mocked values directly instead of having the `post` references.

This benefits us because one no longer needs to figure out why this specific test needs to run before all other tests. 

EDIT: I just found that we already have this test as an actual test further down the line. Feel free to completely remove the `beforeAll` block.